### ISSUE TITLE
[ INFO ] DietPi-Software is already running on another terminal/screen. Please complete or exit, before continuing.

### ISSUE THREAD

**Issue by iCra5h** [MichaIng/DietPi#1443]
After flashing dietpi image to sdcard, i got this message:

[ INFO ] DietPi-Software is already running on another terminal/screen. Please complete or exit, before continuing.

Required Information:
DietPi Version: 6.1
SBC Device: Asus TinkerBoard
SD card used: Sandisk
Distro: Strech ARMv7

**Comment by Fourdee** [comment]
@iCra5h 

Thanks for the report, where did you receive the message?
- on HDMI screen
- SSH

Did you modify any of the optional ```dietpi.txt``` entries to enable automation?

**Comment by iCra5h** [comment]
SSH only. 
I didnt modify any files. Its stock image from site.

**Comment by Fourdee** [comment]
@iCra5h 

If the system is still in this state, please:
- login
- Press CTRL+C
- run ```htop```
- take a screenshot and paste here if possible

**Comment by iCra5h** [comment]
Ok

**Comment by iCra5h** [comment]
![capture](https://user-images.githubusercontent.com/35499812/35639390-bf6d499a-06c2-11e8-8270-49ecafb0caef.PNG)

**Comment by Fourdee** [comment]
@iCra5h 

Thanks, looks like whiptail is cause, i'll try to replicate here. 

~Quick solution for now:~
```
killall -w dietpi-software
killall -w whiptail
killall -w tee
killall -w login
```

Then exit SSH and reconnect to resume setup.

**Comment by Fourdee** [comment]
Confirmed, I'll resolve and re-upload image when done.

**Comment by iCra5h** [comment]
Ok thx so much.

**Comment by Fourdee** [comment]
@iCra5h 

> https://github.com/Fourdee/DietPi/issues/1443#issuecomment-362020797

Does not work, leave it with me.

**Comment by Fourdee** [comment]
Notes:
```
AUTO_SETUP_AUTOMATED=0
```
Automatic login is enabled, strange.

**Comment by iCra5h** [comment]
Yep, doesn't work.

**Comment by Fourdee** [comment]
Fixed image has been uploaded:
http://dietpi.com/downloads/images/DietPi_ASUSTB-ARMv7-Stretch.7z

@iCra5h 
Many thanks for reporting this and assistance 👍 

Tested and completed.

**Comment by iCra5h** [comment]
Thx

**Comment by vnguyen197** [comment]
I have the same issue since updated to v6.20.6, now I can't run dietpi-update either. I'm using raspberry pi 3 Model B

**Comment by DIeMONdello** [comment]
hi. me too v6.20.6 using raspberry pi 3 Model B+

**Comment by DIeMONdello** [comment]
helped me
https://dietpi.com/phpbb/viewtopic.php?f=11&t=5424#p16114

**Comment by vnguyen197** [comment]
Ok, i think this solution is work for me i go to fix this file /DietPi/dietpi/.install_stage from 1 to value of 2 and it works as good as new

**Comment by cmoulliard** [comment]
I also expriment this problem when I flash `DietPi_ROCKPro64-ARMv8-Buster.img` to a SD card and boot the OS using a RockPro64 SBC. I got this screen

<img width="1509" alt="Screenshot 2021-01-19 at 09 30 56" src="https://user-images.githubusercontent.com/463790/105008088-12cd1d00-5a39-11eb-8469-f58c5d2e0850.png">

**Comment by Joulinar** [comment]
Hi, you can cancel this screen once the timer is running, than check for process ID 1359 what it is.

**Comment by cmoulliard** [comment]
> process ID 1359 what it is.
this is the process running the dietpi-login - `root      1359  1356  0 08:28 tty1     00:00:00 /bin/bash /boot/dietpi/dietpi-login`

**Comment by Joulinar** [comment]
you can kill the process, logout and login again

**Comment by cmoulliard** [comment]
I did it and got another beautiful screens ;-)

<img width="1539" alt="Screenshot 2021-01-19 at 10 05 23" src="https://user-images.githubusercontent.com/463790/105012039-fc759000-5a3d-11eb-86ee-ade79dc3ff4d.png">

<img width="1323" alt="Screenshot 2021-01-19 at 10 09 40" src="https://user-images.githubusercontent.com/463790/105012473-7a399b80-5a3e-11eb-8862-6ce1a38b9616.png">

BTW: Is it possible before to boot the first time dietpi with the SD card to disable such a diet-login process in order to launch it manually when we ssh for the first time ?

**Comment by Joulinar** [comment]
> Is it possible before to boot the first time dietpi with the SD card to disable such a diet-login process in order to launch it manually when we ssh for the first time ?

that's the default setting. Usually the first boot is not starting automatically and would need to be set actively. 

That's the respective setting inside `/boot/dietpi.txt`

```
##### Non-interactive Firstrun Setup #####
# On first login, run update, initial setup and software installs without any user input
# - Setting this to "1" is required for below settings to take effect
# - It is HIGHLY recommended to also set CONFIG_BOOT_WAIT_FOR_NETWORK=2, to force infinite wait for network connection during boot and prevent connection timeout errors.
AUTO_SETUP_AUTOMATED=0
```
But in this case, you would not get any `wiptail` menu within the background. As you can see on your first screen, the system is waiting on a user input. Do you have a screen attached?

**Comment by cmoulliard** [comment]
> Do you have a screen attached?

Which screen are you talking here ?

**Comment by Joulinar** [comment]
physical screen directly to the device

**Comment by cmoulliard** [comment]
I cannot plug the RockPro64 SBC to the HDMI screen as I dont have a keyboard attached, just a mouse. This is why I'm using SSH

**Comment by Joulinar** [comment]
it was more the idea to check what is going on 😉

**Comment by MichaIng** [comment]
The `dietpi-login` script is only launched by itself, if you have auto-login enabled or on first boot with `AUTO_SETUP_AUTOMATED=1` set in `dietpi.txt` as Joulinar mentioned. Else you would need to actively enter username and password on the local login prompt. In the latter case, it is only temporarily until first run setup has finished, then after a reboot, the login console halts at the login prompt again. But in that latter case, also the message you see would look different: https://github.com/MichaIng/DietPi/blob/master/dietpi/dietpi-login#L146-L147

I'm currently reviewing the ROCKPro64 image.
... found it, the Armbian images come with a new getty override that force auto-login ... I'll remove that thing.

In your case, please do:
```
killall -SIGKILL -w dietpi-login
rm -Rf /tmp/DietPi-Login
/boot/dietpi/dietpi-login
```

**Comment by cmoulliard** [comment]
> ... found it, the Armbian images come with a new getty override that force auto-login ... I'll remove that thing.
> 
> In your case, please do:
> 
> ```
> killall -SIGKILL -w dietpi-login
> rm -Rf /tmp/DietPi-Login
> /boot/dietpi/dietpi-login
> ```

Many thanks. Will test it...

**Comment by MichaIng** [comment]
Our image preparation script will now assure that this override is present: https://github.com/MichaIng/DietPi/commit/884c2c098a5b3bc37d486efb1b257ab8e95e58f5
ROCKPro64 image is currently uploading, ROCK64 is affected as well, will patch it afterwards and going through some more images.
- [x] ROCKPro64
- [x] ROCK64
- [x] PINE H64
- [x] NanoPC T4
- [x] NanoPi M4V2

**Comment by cmoulliard** [comment]
> killall -SIGKILL -w dietpi-login
> rm -Rf /tmp/DietPi-Login
> /boot/dietpi/dietpi-login

I executed the commands you reported within this ticket but I got this screen again

![Screenshot 2021-01-19 at 12 34 49](https://user-images.githubusercontent.com/463790/105029613-dd352d80-5a52-11eb-81ed-8829731c1ac1.png)

Surprisingly the process `1026` reported within the screenshot do not run anymore 
```
root       487     1  0 11:28 ?        00:00:00 /usr/bin/vmtouch -l /boot/dietpi /boot/dietpi.txt
root       738     1  0 11:28 ?        00:00:00 /usr/sbin/dropbear -p 22 -W 65536
root       847     1  0 11:28 ?        00:00:00 /sbin/dhclient -4 -v -i -pf /run/dhclient.eth0.pid -lf /var/lib/dhcp/dhclient.eth0.leases -I -df /var/
systemd+  1009     1  0 11:29 ?        00:00:00 /lib/systemd/systemd-timesyncd
root      1011     1  0 11:29 ttyS2    00:00:00 /bin/login -f
root      1012     1  0 11:29 tty1     00:00:00 /bin/login -f
root      1021  1012  0 11:29 tty1     00:00:00 -bash
root      1022  1011  0 11:29 ttyS2    00:00:00 -bash
root      1065     1  0 11:29 ttyS2    00:00:00 whiptail --title DietPi-Login --backtitle ROCKPro64 (aarch64) | IP: 192.168.1.50 --msgbox [ INFO ] Die
root      1066     1  0 11:29 tty1     00:00:00 whiptail --title DietPi-Login | File viewer --backtitle ROCKPro64 (aarch64) | IP: 192.168.1.50 --textb
root      1070     2  0 11:33 ?        00:00:00 [kworker/5:1-mm_percpu_wq]
root      1071   738  0 11:33 ?        00:00:00 /usr/sbin/dropbear -p 22 -W 65536
root      1072  1071  0 11:33 pts/0    00:00:00 -bash
root      1097     2  0 11:33 ?        00:00:00 [kworker/5:3-events]
root      1106     2  0 11:34 ?        00:00:00 [kworker/2:0-mm_percpu_wq]
root      1118     2  0 11:34 ?        00:00:00 [kworker/u12:2-events_unbound]
root      1127     2  0 11:34 ?        00:00:00 [kworker/0:0-events]
root      1154     2  0 11:36 ?        00:00:00 [kworker/1:0-mm_percpu_wq]
```

**Comment by MichaIng** [comment]
Ah sorry I forgot the flag file itself, an of course to remove the autologin:
```
rm /tmp/.dietpi-login_firstrun_setup.pid
rm -R  /etc/systemd/system/getty@.service.d
/boot/dietpi/dietpi-login
```
**EDIT: All affected images have been patched.**

**Comment by cmoulliard** [comment]
> EDIT: All affected images have been patched.

How can I test a patched image ?

**Comment by Joulinar** [comment]
try to download new image for your device https://dietpi.com/downloads/images/

**Comment by cmoulliard** [comment]
New image appears under `testing` dir --> `https://dietpi.com/downloads/images/testing/DietPi_RPi-ARMv8-Buster.7z`
and not `https://dietpi.com/downloads/images/` ...

**Comment by Joulinar** [comment]
you need to use RockPro64 image an not RPi. At least you told about using RockPro64 above https://github.com/MichaIng/DietPi/issues/1443#issuecomment-762685447 😉 

https://dietpi.com/downloads/images/DietPi_ROCKPro64-ARMv8-Buster.7z

**Comment by cmoulliard** [comment]
Sorry, wrong copy/paste. 

I was surprised to discover on the server that image dates were outdated and why I looked under `testing` sub-folder
```
Example for RockPro64
DietPi_ROCKPro64-ARMv8-Buster.7z | 2020-12-23 01:37 | 122M
```

**Comment by Joulinar** [comment]
Probably a cache refresh is needed on the browser

`DietPi_ROCKPro64-ARMv8-Buster.7z	2021-01-19 12:34	122M`

**Comment by MichaIng** [comment]
Sorry for the image update confusion. Since I didn't change anything but removed the autologin override, I replaced the regular downloads directly. The directory index may not represent the correct timestamps since we have two layers of cache: Cloudflare up to 1 day and browser cache up to 1 day 😅. I cleared Cloudflare cache for the images themselves but not for the directory index. I.e. the regular downloads (the one Joulinar linked above) are the fixed images already.
